htdocs/product/card.php

```diff
@@ -1211,7 +1211,13 @@
 			// Accountancy_code_buy
 			print '<tr><td>'.$langs->trans("ProductAccountancyBuyCode").'</td>';
 			print '<td>';
-			print $formaccounting->select_account(GETPOST('accountancy_code_buy', 'alpha'), 'accountancy_code_buy', 1, null, 1, 1, '');
+            if($type == 0)
+            {
+                $accountancy_code_buy = (GETPOST('accountancy_code_buy', 'alpha')?(GETPOST('accountancy_code_buy', 'alpha')):$conf->global->ACCOUNTING_PRODUCT_BUY_ACCOUNT);
```

Comment:
The constant ACCOUNTING_PRODUCT_BUY_ACCOUNT is the account that will be used in accounting module when the accountancy code of product is not defined. So we should not have it suggested here, product accounting code should stay empty if we don't want to define a given code. Otherwise, changing the setup will have not effect.

Reply:
then why we preselect Accountancy_code_sell_export, Accountancy_code_sell_intra, Accountancy_code_sell (all other accountancy code)